hibernate-core/src/main/java/org/hibernate/metamodel/model/convert/internal/NamedEnumValueConverter.java

```diff
@@ -25,12 +27,17 @@
  * @author Steve Ebersole
  */
 public class NamedEnumValueConverter<E extends Enum> implements EnumValueConverter<E,String>, Serializable {
-	private static final Logger log = Logger.getLogger( NamedEnumValueConverter.class );
 
 	private final EnumJavaTypeDescriptor<E> enumJavaDescriptor;
 
+	private final transient ValueExtractor<E> valueExtractor;
+
+	private final transient ValueBinder<String> valueBinder;
```

Comment:
I see you marked them as transient. Do we expect the converter to be serialized (I see it's `Serializable` so this part is true) and then reused (wondering about this specific part)?

If so, we would need to implement a `readObject()` method calling the default `readObject()` and then reinitializing the two fields. Obviously, you would have to remove the `final` qualifier in this case.

Reply:
In 5.x the converter might be serialized, so this part is fine.

Seeing the null getting passed along, I wonder if we want to change the signature of `EnumValueConverter#readValue` and `EnumValueConverter#writeValue` to pass along the Session

Reply:
@sebersole 

About `In 5.x the converter might be serialized, so this part is fine.`, you're saying that adding a `readObject()` method is unnecessary?

I don't know if we expect the converter to be reused after deserialization or if we just want it to be serializable?

About adding the session as a parameter, it seems like a good idea. `EnumValueConverter` is an SPI but I don't expect it to be implemented elsewhere.

Reply:
I don't understand how changing the signature from `WrapperOptions` to `Session` would solve the null passing problem. Either way, we won't have a reference to the `Session` in the `readValue` method:

```
@Override
public E readValue(ResultSet resultSet, String name) throws SQLException {
	return valueExtractor.extract( resultSet, name, null );
}
```

Reply:
Which is why I said to pass it in to the converter.  Which is in fact exactly what we have done in 6 for just this reason.  So:

````
    @Override
    E readValue(
            ResultSet resultSet, 
            String name,  
            SharedSessionContractImplementor session) throws SQLException {
	return valueExtractor.extract( resultSet, name, session );
}
````

Reply:
@gsmet 

> About `In 5.x the converter might be serialized, so this part is fine.`, you're saying that adding a `readObject()` method is unnecessary?

I guess it depends on where the converter is serialized "from".  

If it is serialized as part of the SessionFactory (which I think is true) then the readObject is unnecessary because we never *really* serialize the SessionFactory - we simply resolve it readResolve (though in retrospect, I think Externalizable would have been better).

It is a different story if it ever gets serialized as part of a Session (outside of the Session's reference to SF).  There it would need the readObject/


> About adding the session as a parameter, it seems like a good idea. `EnumValueConverter` is an SPI but I don't expect it to be implemented elsewhere.

Right.  The "API" would be JPA's AttributeConverter.